controllers/topology/internal/log/log.go

```diff
@@ -76,40 +76,44 @@ type topologyReconcileLogger struct {
 // WithObject adds to the logger information about the object being modified by reconcile, which in most case it is
 // a resources being part of the Cluster by reconciled.
 func (l *topologyReconcileLogger) WithObject(obj client.Object) Logger {
-	l.Logger = l.Logger.WithValues(
-		"object groupVersion", obj.GetObjectKind().GroupVersionKind().GroupVersion().String(),
-		"object kind", obj.GetObjectKind().GroupVersionKind().Kind,
-		"object", obj.GetName(),
-	)
-	return l
+	return &topologyReconcileLogger{
+		Logger: l.Logger.WithValues(
```

Comment:
Just fyi. klog was using a similar way in earlier versions and it had bugs that some fields of the logger struct were not copied. In our case this cannot happen (right now) as we don't have other fields. But we should be careful in case we add new fields in the future.

An alternative would be:
```
func (l topologyReconcileLogger) WithObject(obj client.Object) Logger {
	l.Logger = l.Logger.WithValues(
		"object groupVersion", obj.GetObjectKind().GroupVersionKind().GroupVersion().String(),
		"object kind", obj.GetObjectKind().GroupVersionKind().Kind,
		"object", obj.GetName())
	return &l
}
```
(cf. https://github.com/kubernetes/klog/blob/main/klogr/klogr.go#L206)

Essentially this works as it has a value receiver, so the logger is copied before even calling the func.

I personally prefer your current implementation as I think a value receiver is too implicit. But I just wanted to bring it up in case you have another opinion / as background info.

Reply:
Thanks for the context, let's go with the more explicit option and reconsider if the need arise